planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp

```diff
@@ -523,12 +514,37 @@ TransformStamped FreespacePlannerNode::getTransform(
 
 void FreespacePlannerNode::initializePlanningAlgorithm()
 {
-  if (node_param_.planning_algorithm == "astar") {
-    algo_.reset(new AstarSearch(planner_common_param_, astar_param_));
+  // Extend robot shape
+  freespace_planning_algorithms::VehicleShape extended_vehicle_shape = vehicle_shape_;
+  const double margin = node_param_.collision_margin;
+  extended_vehicle_shape.length += margin;
+  extended_vehicle_shape.width += margin;
+  extended_vehicle_shape.base2back += margin / 2;
+
+  const auto planner_common_param = getPlannerCommonParam();
+
+  const auto algo_name = node_param_.planning_algorithm;
+
+  // initialize specified algorithm
+  if (algo_name == "astar") {
+    AstarParam p;
+    p.only_behind_solutions = declare_parameter("astar.only_behind_solutions", false);
+    p.use_back = declare_parameter("astar.use_back", true);
+    p.distance_heuristic_weight = declare_parameter("astar.distance_heuristic_weight", 1.0);
+    algo_ = std::make_unique<AstarSearch>(planner_common_param, extended_vehicle_shape, p);
+
+  } else if (algo_name == "rrtstar") {
+    RRTStarParam p;
+    p.enable_update = declare_parameter("rrtstar.enable_update", true);
+    p.use_informed_sampling = declare_parameter("rrtstar.use_informed_sampling", true);
+    p.mu = declare_parameter("rrtstar.neighbour_radius", 8.0);
+    p.max_planning_time = declare_parameter("rrtstar.max_planning_time", 150);
+    p.margin = declare_parameter("rrtstar.margin", 0.1);
+    algo_ = std::make_unique<RRTStar>(planner_common_param, extended_vehicle_shape, p);
```

Comment:
@HiroIshida can you move to each constructor such as RRTStar and AstarSearch?
You can doit by passing rclcpp::Node to the constructor.
https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_velocity_planner/include/scene_module/scene_module_interface.hpp#L111

Reply:
@yukkysaito 
Thanks for your comment. I tried to implement following your idea for an hour (implementing algorithm's constructor that takes node as input), but It is bit difficult to implement it in clean way (causing many copy of same fraction of code inside due to already-exisiting constructors. 

So, I am considering the following implementation (pseudo code in python).  By the following implementation, we can 100% reuse the already-exists constructor of each algorithms. 

Sorry for confusing you. 

```python
from abc import ABC, abstractmethod

# in abstract_algorithm.hpp
class AbstractPlannerCreator(ABC):

    def create(node, vehicle_shape):
        planner_common_param = getPlannerCommonParam(node)

    @abstractmethod
    def _create(node, planner_common_param, vehicle_shape):  # this can be named create in c++ thanks to function overload
        # algorithm specific implementation
        pass

# in astar_search.hpp
class AstarSearchCreator(AbstractPlannerCreator):

    def _create(node, planner_common_param, vehicle_shape) -> AstarSearch:
        astar_param = getAstarParam(node)
        return AstarSearch(planner_common_param, astar_param, vehicle_shape)

# in rrtstar.hpp
class RRTStarCreator(AbstractPlannerCreator):

    def _create(node, planner_common_param, vehicle_shape) -> RRTStar:
        rrtstar_param = getRRTStarParam(node)
        return RRTStar(planner_common_param, astar_param, vehicle_shape)


if __name__ == "__main__":
    # in c++, the following line will be replaced by unique_ptr<AstarSearch> cretator = unique_ptr<AbstractPlannerCreator>(args);
    algo_creator = RRTStarCreator(node, vehicle_shape)
    algo = algo_creator.create()
```